editing_and_imputation/editing/totals_components/technical_specification.md

```diff
@@ -0,0 +1,316 @@
+the total or components.
+
+To do this our first step is to verify if the following is true.
+
+```bash
+    sum_of_components = 0
+```
+
+and
+
+```bash
+    predictive_value > 0
+```
+
+If this is the case then we have TCC Marker = S and the method
+stops. We then write an output.
+
+If however the above is false we move onto stage 4
+
+### 5.3 Check Sum(components) and Predictive Value (Stage 4)
+
+Stage 4 is where we check the sum of the components and the
+predictive value to determine whether there is a difference that
+may require correction.
+
+The initial part requires us to determine the absolute difference
+between the total and sum of the components.
+
+```bash
+    |predictive_value - sum_of_components| = absolute_difference_threshold.
+```
+
+Note: The computed absolute difference needs to be available
+so that it can be output when results are returned. It is also
+important to understand we use the python
+Decimal() method to apply the defined precision value
+to the floating values for this stage.
+This determines the accuracy of out outputs.
+
+We now determine if the absolute difference between the sum of
+the components and the predictive is zero then the method stops
+with an indication of no correction.
+
+If true, we have a TCC Marker = N meaning we have no correction
+and the method stops with an output written.
+
+Else we move onto the next stage which is where we determine
+an error detection method.
+
+### 5.4 Determine Error Detection Method (Stage 5)
+
+Stage 5 determines whether the detected difference in the
+provided components and totals falls into the absolute or percentage
+difference thresholds specified and therefore needs to be
+automatically correct or the values need to be manually corrected.
+
+When only Absolute Difference Threshold is set to 0 or above
+the threshold is checked against the absolute value calculated
+in step 4. We complete this action in section 5.4.1.
+
+When only Percentage Difference Threshold is set to 0 or above
+the threshold is checked against the predictive total. The percentage
+threshold is the sum of the received components plus and minus the
+Percentage Difference Threshold specified. If this threshold is not
+satisfied then we require manual editing and the method stops.
+
+### 5.4.1 Check Absolute Difference Threshold (Stage 5a)
+
+The absolute difference between the target total and
+the components must be less than or equal to the absolute difference threshold.
+
+When the Absolute Difference Threshold check indicates the
+difference needs to be automatically corrected the method
+continues to stage 6 in section 5.5.
+
+When the Absolute Difference Threshold check indicates the
+correction needs to be manually applied the method will check
+if a Percentage Difference Threshold needs to be checked.
+
+Note: Before we leave this stage we need to check
+that zero error condition 3 is satisfied i.e. If
+target total = 0 and components sum > 0 and amend total = TRUE:
```

Comment:
You should write this in the same terms you use elsewhere e.g total_value, sum_of_components etc